pkg/controller.v1/mpi/mpijob_controller.go

```diff
@@ -176,9 +176,10 @@ func (jc *MPIJobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 }
 
 // SetupWithManager sets up the controller with the Manager.
-func (jc *MPIJobReconciler) SetupWithManager(mgr ctrl.Manager) error {
+func (jc *MPIJobReconciler) SetupWithManager(mgr ctrl.Manager, controllerThreads int) error {
```

Comment:
Can we or should we introduce a way to keep the `SetupWithManager` API, may be extending the configuration struct is better than extending/changing API. 

Is that make sense to introduce `MaxConcurrentReconciles` config to `common.JobController` ?

What's your opinion? @johnugeorge

Reply:
@kuizhiqing Yes. But I think, we need more refactoring. eg: [NewReconciler](https://github.com/kubeflow/training-operator/blob/master/pkg/controller.v1/tensorflow/tfjob_controller.go#L94) can take [common.JobControllerConfiguration ](https://github.com/kubeflow/training-operator/blob/master/pkg/controller.v1/tensorflow/tfjob_controller.go#L112) as argument  and we can add `MaxConcurrentReconciles` to `common.JobControllerConfiguration`.  

We can do this in a separate PR as well.

Reply:
@johnugeorge yes, I agree with that.